mne/viz/evoked.py

```diff
@@ -1448,6 +1449,10 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
     elif not isinstance(evokeds, dict):
         evokeds = dict((str(ii + 1), evoked)
                        for ii, evoked in enumerate(evokeds))
+    for cond in evokeds.keys():
+        if not isinstance(cond, str):
+            t = type(cond)
+            raise TypeError("Conditions must be str, not {}.".format(t))
```

Comment:
we prefer `%` for simplicity, just `'Conditions must be string, not %s' % (type(cond),)`